src/gs-shell.c

```diff
@@ -782,6 +788,37 @@ gs_shell_reload_cb (GsPluginLoader *plugin_loader, GsShell *shell)
 	}
 }
 
+static void
+gs_shell_reload_and_select_removable_media_category (GsShell *shell,
+						     gboolean reload_overview,
+						     gboolean select_category,
+						     const gchar *category_name)
+{
+	/* using NULL for this below indicates it should not be selected */
+	const char *cat_name = (select_category ? category_name : NULL);
+
+	if (!reload_overview) {
+		reload_overview_and_select_category (shell, cat_name);
+	} else {
+		GsCategory *category = side_filter_get_category_by_name (shell, cat_name);
+		if (category == NULL) {
+			reload_overview_and_select_category (shell, cat_name);
+		} else {
+			gs_shell_change_mode (shell, GS_SHELL_MODE_CATEGORY,
+					      (gpointer) category, TRUE);
+		}
+	}
+	gs_shell_activate (shell);
+}
+
+static void
+gs_shell_copy_dests_notify_cb (GsPluginLoader *plugin_loader,
+			       GParamSpec *pspec,
+			       GsShell *shell)
+{
+	gs_shell_reload_and_select_removable_media_category (shell, TRUE, FALSE, "usb");
```

Comment:
Yeah, this is a bit cryptic, `gs_shell_reload_and_select_removable_media_category (shell, TRUE, NULL);` would be better. But actually only reloading the shell or selecting the removable category would even be clearer.

Reply:
This also goes away in recent simplification